### PR TITLE
Allow an identity to express its locale preference

### PR DESCRIPTION
**Pull request by rjkip**
Requires DB zap

### CODE REVIEW

**Comment by rjkip** [comment]
:sparkler: @DRvanR

File: src/Surfnet/Stepup/Identity/Entity/ConfigurableSettings.php [hunk 1]
```diff
@@ -56,4 +87,19 @@ public function createNewEmailVerificationWindow()
             DateTime::now()
         );
     }
+
+    /**
+     * @param Locale $locale
+     * @return bool
+     */
+    public function isValidLocale(Locale $locale)
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Identity/Entity/ConfigurableSettings.php]: `isKnownLocale`? The locale may be valid, but not known by the system...

File: src/Surfnet/StepupMiddleware/MiddlewareBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -42,6 +42,17 @@ public function getConfigTreeBuilder()
                             )
                         ->end()
                     ->end()
+                    ->arrayNode('locales')
+                        ->info('An array of locales that are available in the domain')
```
Comment (DRvanR) [inline src/Surfnet/StepupMiddleware/MiddlewareBundle/DependencyInjection/Configuration.php]: Consider getting these from the `locales` parameter :smile:

File: src/Surfnet/Stepup/Identity/Entity/ConfigurableSettings.php (hunk 1 above)
Comment (DRvanR) [inline src/Surfnet/Stepup/Identity/Entity/ConfigurableSettings.php]: `isSupportedLocale`?

**Comment by rjkip** [comment]
:sparkles: @DRvanR

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php [hunk 3]
```diff
@@ -217,4 +227,28 @@ public function handleRevokeRegistrantsSecondFactorCommand(RevokeRegistrantsSeco
...
+        $this->repository->save($identity);
+    }
+
+    /**
+     * @param Locale $locale
+     */
+    private function assertIsValidLocale(Locale $locale)
+    {
+        if (!$this->configurableSettings->isSupportedLocale($locale)) {
+            throw new DomainException(
```
Comment (DRvanR) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php]: `InvalidCommandException`?

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Exception/DomainException.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\CommandHandlingBundle\Exception;
+
+use DomainException as CoreDomainException;
+
+class DomainException extends CoreDomainException implements Exception
```
Comment (DRvanR) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Exception/DomainException.php]: Not sure this should exist...

**Comment by DRvanR** [comment]
One minor (combined :wink:) nitpick, bar that :+1:

**Pull request merged**